### PR TITLE
Analytical_TTL Replace Functionality

### PR DESCRIPTION
**Pull request by JericHunter**
# Description
This PR is for the replace container functionality for the analytical storage ttl feature.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/azure/cosmos/aio/_database.py [hunk 1]
```diff
@@ -437,6 +437,9 @@ async def replace_container(
         :paramtype response_hook: Callable[[Dict[str, str], Dict[str, Any]], None]
         :raises ~azure.cosmos.exceptions.CosmosHttpResponseError: Raised if the container couldn't be replaced.
             This includes if the container with given id does not exist.
+        :keyword analytical_storage_ttl: Analytical store time to live (TTL) for items in the container.  A value of
```
Comment (simorenoh) [inline sdk/cosmos/azure-cosmos/azure/cosmos/aio/_database.py:440]: please add keyword type the way we have types for other keywords and variables

File: sdk/cosmos/azure-cosmos/azure/cosmos/database.py [hunk 2]
```diff
@@ -487,6 +487,9 @@ def replace_container(
         :keyword Callable response_hook: A callable invoked with the response metadata.
         :raises ~azure.cosmos.exceptions.CosmosHttpResponseError: Raised if the container couldn't be replaced.
             This includes if the container with given id does not exist.
+        :keyword analytical_storage_ttl: Analytical store time to live (TTL) for items in the container.  A value of
```
Comment (simorenoh) [inline sdk/cosmos/azure-cosmos/azure/cosmos/database.py:490]: same here

File: sdk/cosmos/azure-cosmos/test/test_analytical_ttl.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
+# The MIT License (MIT)
+# Copyright (c) 2014 Microsoft Corporation
```
Comment (simorenoh) [inline sdk/cosmos/azure-cosmos/test/test_analytical_ttl.py:2]: 2022

File: sdk/cosmos/azure-cosmos/test/test_analytical_ttl_asnyc.py [hunk 4]
```diff
@@ -0,0 +1,297 @@
+# The MIT License (MIT)
+# Copyright (c) 2014 Microsoft Corporation
```
Comment (simorenoh) [inline sdk/cosmos/azure-cosmos/test/test_analytical_ttl_asnyc.py:2]: 2022 and please double check the naming on the file :)

**Review by simorenoh — APPROVED** [review]
only some very small comments, awesome work Jeric!

**Comment by simorenoh** [comment]
OH and please add relevant Changelog entry and remove the limitation from the README

**Review by ealsur — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,7 @@
 
 #### Features Added
 - GA release of integrated cache functionality. For more information on integrated cache please see [Azure Cosmos DB integrated cache](https://docs.microsoft.com/azure/cosmos-db/integrated-cache).
+- GA release of analytical ttl replace functionality. For more information on analytical ttl please see [Azure Cosmos DB analytical store](https://docs.microsoft.com/azure/cosmos-db/analytical-store-introduction).
```
Comment (simorenoh) [inline sdk/cosmos/azure-cosmos/CHANGELOG.md:7]: no need to mark this as GA - GA is when we release functionality that was previously in beta. 
```suggestion
- Added ability to replace analytical ttl on containers. For more information on analytical ttl please see [Azure Cosmos DB analytical store](https://docs.microsoft.com/azure/cosmos-db/analytical-store-introduction).
```

Comment (JericHunter) [reply sdk/cosmos/azure-cosmos/CHANGELOG.md]: got ya

File: sdk/cosmos/azure-cosmos/azure/cosmos/aio/_database.py [hunk 6]
```diff
@@ -437,6 +437,9 @@ async def replace_container(
         :paramtype response_hook: Callable[[Dict[str, str], Dict[str, Any]], None]
         :raises ~azure.cosmos.exceptions.CosmosHttpResponseError: Raised if the container couldn't be replaced.
             This includes if the container with given id does not exist.
+        :keyword int analytical_storage_ttl: Analytical store time to live (TTL) for items in the container.  A value of
+            None leaves analytical storage off and a value of -1 turns analytical storage on with no TTL.  Please
+            note that analytical storage can only be enabled on Synapse Link enabled accounts.
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/azure/cosmos/aio/_database.py:442]: What happens if I try to set this value on a non-Synapse-link-enabled account?

File: sdk/cosmos/azure-cosmos/test/test_analytical_ttl.py [hunk 7]
```diff
@@ -0,0 +1,296 @@
...
+            analytical_storage_ttl=ttl)
+        created_collection_properties = created_collection.read()
+        self.assertEqual(created_collection_properties['analyticalStorageTtl'], ttl)
+
+        collection_id = 'test_ttl_values4' + str(uuid.uuid4())
+        ttl = -10
+
+        # -10 is an unsupported value for ttl. Valid values are -1 or a non-zero positive 32-bit integer value
+        self.__AssertHTTPFailureWithStatus(
+            StatusCodes.BAD_REQUEST,
```
Comment (annatisch) [inline sdk/cosmos/azure-cosmos/test/test_analytical_ttl.py:108]: A generic "bad request" error is not a good experience - what's the actual error message here? Would it tell users why the failure occurred and how to fix it?

**Pull request merged**